### PR TITLE
OOP school library - add basic UI : Pull Request

### PR DESCRIPTION
**Pull request by belaymit**
## Description
**_In this project,_** I created a form of UI for my school library app. This way it can be invoked as an executable and not something I use in IRB exclusively.

## Project requirements

- Create a ``app.rb`` file that will serve as your console app entry-point. It should have methods that do the following:

- [ ]     List all books.
- [ ]     List all people.
- [ ]     Create a person (teacher or student, not a plain Person).
- [ ]     Create a book.
- [ ]     Create a rental.
- [ ]     List all rentals for a given person id.

-  In your ``main.rb`` define the entry point, this will be a method called main that is invoked at the end of your file. This method should do the following:

- [ ]     Present the user with a list of options to perform.
- [ ]     Lets users choose an option.
- [ ]     If needed, ask for parameters for the option.
- [ ]     Have a way to quit the app.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require_relative './app'
+
```
Comment (Mihndim2020) [inline main.rb:2]: - [ ] I am getting an error when I try to run your code, checkout the screenshot below and kindly fix the issue. Please make sure you app is running according to the project [requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_ui.md) before you submit for another review.  

ruby error

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi @belaymit, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark:  Clean and Organized code
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  def create_rental(book_id, person_id, rental_date, return_date)
+    # code to create a rental
+  end
+
+  def load_rentals(person_id)
+    # code to list all rentals for a given person id
+  end
+end
```
Comment (mckent05) [inline app.rb]: - Please implement all tasks required for this [milestone] (https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_ui.md). Please implement the code required to carry out each of those tasks. Please fix this 🙏 . Currently your application only prints `display books`.

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @belaymit, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost there


 ## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your code.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/classes/person.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@ class Person < Nameable
   attr_accessor :name, :age, :rentals
   attr_reader :id, :type
 
-  def initialize(age, name = 'Unknown', parent_permission: true)
+  def initialize(age, name, parent_permission: true)
```
Comment (ghost) [inline src/classes/person.rb:8]: - [ ] Good work displaying the options, but when I try to create a person, an error occurs. Kindly check it out.

image

image

**Review by ghost — CHANGES_REQUESTED** [review]
Hi @belaymit , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ Correct use of Gitflow
✅ No linter error


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**